packages/common/core-utils/src/assert.ts

```diff
@@ -5,13 +5,25 @@
 
 /**
  * A browser friendly assert library.
- * Use this instead of the 'assert' package, which has a big impact on bundle sizes.
+ *
  * @param condition - The condition that should be true, if the condition is false an error will be thrown.
  * Only use this API when `false` indicates a logic error in the problem and thus a bug that should be fixed.
  * @param message - The message to include in the error when the condition does not hold.
  * A number should not be specified manually: use a string.
  * Before a release, policy-check should be run, which will convert any asserts still using strings to
  * use numbered error codes instead.
+ * @remarks
+ * Use this instead of the node 'assert' package, which requires polyfills and has a big impact on bundle sizes.
+ *
+ * Asserts using this API will be included in all configurations: there is no option to disable or optimize them out.
+ * Thus this API is suitable for detecting conditions that should terminate the application and produce a useful diagnostic message.
+ * It can be used to ensure bad states are detected early and avoid data corruption or harder to debug errors.
+ *
+ * In cases where the assert is very unlikely to have an impact on production code but is still useful as documentation and for debugging, consider using `debugAssert` instead
+ * to optimize bundle size.
+ * @privateRemarks
+ * This should be deprecated (as a non internal API) then moved to purely internal.
```

Comment:
Does anyone know if we have any work tracking migrating external users off of this? In the interim, should we update the function summary docs to indicate that external users should not use it?

Reply:
We do plan to eventually replace all alpha APIs (both legacy and non-legacy ones), so the alpha tag kind of puts this in a list, but that work is long term. I'm not aware of any more detailed tracking, like a list of alpha APIs we want to make internal via deprecation.

Reply:
I have added a commented about the intended internalness.